sympy/core/numbers.py

```diff
@@ -649,49 +660,122 @@ class Float(Number):
+    by passing it to Rational (see also discusion below on SymPy Float
+    and Python float):
+
+    >>> from sympy import Rational
+    >>> Rational(Float(1/8))
+    1/8
+    >>> Rational(Float(1/10, 1))  # the 1 digit approximation
+    51/512
+    >>> Rational(Float(1/10, 2))  # the 2 digit approximation
+    819/8192
+
+    While operations involving Rationals are exact, Floats interact under
+    the constraints of the binary denominator and limited precision.
+    Consider the 2 digit approximations for 1/10 and 1/3:
+
+    >>> a, b = Float(1/10, 2), Float(1/3, 2); (a, b)
+    (0.1, 0.33)
+
+    The underlying rational approximations are
+
+    >>> [Rational(i) for i in _]
+    [819/8192, 683/2048]
 
-    >>> approx, exact = Float(.1, 1), Float(.125, 1)
+    The exact sum of them is
 
-    For calculation purposes, evalf needs to be able to change the precision
-    but this will not increase the accuracy of the inexact value. The
-    following is the most accurate 5-digit approximation of a value of 0.1
-    that had only 1 digit of precision:
+    >>> sum(_)
+    3551/8192
 
-    >>> approx.evalf(5)
-    0.099609
+    But that result would require more precision than was specified so
+    the actual result is different:
 
-    By contrast, 0.125 is exact in binary (as it is in base 10) and so it
-    can be passed to Float or evalf to obtain an arbitrary precision with
-    matching accuracy:
+    >>> Rational(Float(_, 2))
+    111/256
 
-    >>> Float(exact, 5)
-    0.12500
-    >>> exact.evalf(20)
-    0.12500000000000000000
+    So when we add ``a`` and ``b``, that is the underlying value that
+    is obtained:
 
-    Trying to make a high-precision Float from a float is not disallowed,
-    but one must keep in mind that the *underlying float* (not the apparent
-    decimal value) is being obtained with high precision. For example, 0.3
-    does not have a finite binary representation. The closest rational is
-    the fraction 5404319552844595/2**54. So if you try to obtain a Float of
-    0.3 to 20 digits of precision you will not see the same thing as 0.3
-    followed by 19 zeros:
+    >>> a + b, Rational(a + b)
+    (0.43, 111/256)
+
+    So although Floats are using underlying rationals, those rationals
+    combine under constraints of precision. This makes them
+    computationally efficient but less exact than Rationals.
+
+    == Floats are rational...but not Rational ==
+
+    Integers, Rationals, and Floats are all SymPy Numbers. While all
+    Numbers are rational, the attribute `is_Rational` is True only
+    for Integers and Rational. Other lower-case attributes may
+    also be true for Floats:
+
+    >>> f1 = Float(1)
+    >>> f1.is_Rational, f1.is_rational, f1.is_integer, f1.is_odd
+    (False, True, True, True)
+    >>> r = Float(0.1)
+    >>> r.is_rational, r.is_integer, r.is_odd
+    (True, False, False)
+
+    == SymPy Float from Python float ==
```

Comment:
Not sure how this sort of header will render in Sphinx.

Reply:
I redid these using `----` underlining